api/src/play_asset_pack_manager.gd

```diff
@@ -185,6 +284,25 @@ func get_asset_pack_state(pack_name: String) -> PlayAssetPackStateRequest:
 	_plugin_singleton.getPackStates([pack_name], signal_id)
 	return return_request
 
+# -----------------------------------------------------------------------------
+# Requests to download the specified asset pack.
+# -----------------------------------------------------------------------------
+func fetch_asset_pack(pack_name: String) -> PlayAssetPackFetchRequest:
+	var return_request = PlayAssetPackFetchRequest.new(pack_name)
+	var signal_id = _request_tracker.register_request(return_request)
+	
+	# Update mapping of pack_name to request object, so that assetStateUpdated global signal	
+	# can be correctly routed to this request object.	
+	_play_asset_pack_manager_mutex.lock()	
+	if _asset_pack_to_request_map.has(pack_name):	
+		_asset_pack_to_request_map[pack_name].append(return_request)	
```

Comment:
Do we need to create a new return_request here? Is it possible to return _asset_pack_to_request_map[pack_name][0] aka. the existing request?

Reply:
It could simplify the handling code to only have one request per pack active at one time.

Reply:
Yeah I think this is a good call. I refactored this ```_asset_pack_to_request_map``` to map to single request object instead a list of request objects.